### PR TITLE
cluster: Automatically create machine ACLs for public connections

### PR DESCRIPTION
**Pull request by kklin**
Before, the only notion of machine ACLs were Admin ACLs, which allowed
all traffic to the cluster. Making applications publicly accessible
using Admin ACLs posed a security risk because sensitive ports such as
those for SSH, Docker Swarm, and our Minion GRPC were also made open.

We now automatically generate and enforce ACLs based on the public
connections defined in the stitch.

### CODE REVIEW

File: cluster/provider/awsSpot.go [hunk 1]
```diff
@@ -284,47 +284,47 @@ func (clst amazonCluster) List() ([]Machine, error) {
 				Provider: db.Amazon,
```
Comment (ejj) [inline cluster/provider/awsSpot.go]: How thoroughly have you tried this out?  I think I didn't include them for a reason, but I can't quite remember what it is.

Anyways, if you're confident it works, I'm fine with it.

File: cluster/provider/awsSpot.go [hunk 2]
```diff
@@ -468,7 +469,7 @@ func (clst *amazonCluster) GetCreateSecurityGroup(session EC2Client) (
 	return *csgResp.GroupId, nil, nil
 }
 
-func (clst *amazonCluster) SetACLs(acls []string) error {
+func (clst *amazonCluster) SetACLs(acls []ACL) error {
```
Comment (ejj) [inline cluster/provider/awsSpot.go]: Is this the right patch for this change?  Might be cleaner to do it one earlier.

File: cluster/provider/awsSpot.go (hunk 1 above)
Comment (kklin) [inline cluster/provider/awsSpot.go]: We used to include them, but this patch removed them while trying to decrease indentation I think: https://github.com/NetSys/quilt/commit/19d955efd62032014336dd8d7e5af573bbaf7578#diff-97341d2e1fc06dde22fa710e89b4c1a4L281

File: cluster/provider/awsSpot.go [hunk 3]
```diff
@@ -476,95 +477,44 @@ func (clst *amazonCluster) SetACLs(acls []string) error {
...
-					_, err = session.RevokeSecurityGroupIngress(
-						options)
-					if err != nil {
-						return err
-					}
-				} else {
-					groupIngressExists = true
-				}
+		if len(rangesToAdd) != 0 {
+			logACLs(true, rulesToRemove)
```
Comment (kklin) [inline cluster/provider/awsSpot.go]: Not sure when I fixed this, but it looks correct now..

File: cluster/provider/awsSpot.go (hunk 2 above)
Comment (kklin) [inline cluster/provider/awsSpot.go]: So, have a new commit that changes the interface to be `SetACLs([]ACL)`, and make the final commit just create the machine ACLs automatically?

Comment (ejj) [inline cluster/provider/awsSpot.go]: Yep that's exactly what I mean.

**Comment by ejj** [comment]
Also this merge conflicts with master.

**Comment by kklin** [comment]
Addressed comments

**Pull request merged**